batavia/VirtualMachine.js

```diff
@@ -755,95 +755,161 @@ VirtualMachine.prototype.create_traceback = function() {
+            var opcode_start_pos = pos
+
+            var opcode = code.co_code.val[pos++]
+
+            // next opcode has 4-byte argument effectively.
+            if (opcode === dis.EXTENDED_ARG) {
+                lo = code.co_code.val[pos++]
+                hi = code.co_code.val[pos++]
+                extra = (lo << 16) | (hi << 24)
+                // emulate four NOPs
+                unpacked_code[opcode_start_pos] = {
+                    'opoffset': opcode_start_pos,
+                    'opcode': dis.NOP,
+                    'op_method': this.dispatch_table[dis.NOP],
+                    'args': [],
+                    'next_pos': pos
+                }
+                unpacked_code[opcode_start_pos + 1] = {
+                    'opoffset': opcode_start_pos + 1,
+                    'opcode': dis.NOP,
+                    'op_method': this.dispatch_table[dis.NOP],
+                    'args': [],
+                    'next_pos': pos
                 }
-            } else if (opcode in dis.hasname) {
-                args = [code.co_names[intArg]]
-            } else if (opcode in dis.hasjrel) {
-                args = [pos + intArg]
-            } else if (opcode in dis.hasjabs) {
-                args = [intArg]
-            } else if (opcode in dis.haslocal) {
-                args = [code.co_varnames[intArg]]
+                unpacked_code[opcode_start_pos + 2] = {
+                    'opoffset': opcode_start_pos + 2,
+                    'opcode': dis.NOP,
+                    'op_method': this.dispatch_table[dis.NOP],
+                    'args': [],
+                    'next_pos': pos
+                }
+                unpacked_code[opcode_start_pos + 3] = {
+                    'opoffset': opcode_start_pos + 3,
+                    'opcode': dis.NOP,
+                    'op_method': this.dispatch_table[dis.NOP],
+                    'args': [],
+                    'next_pos': pos
+                }
+                continue
+            }
+
+            if (opcode < dis.HAVE_ARGUMENT) {
+                args = []
             } else {
-                args = [intArg]
+                lo = code.co_code.val[pos++]
+                hi = code.co_code.val[pos++]
+                var intArg = lo | (hi << 8) | extra
+                extra = 0 // use extended arg if present
+
+                if (opcode in dis.hasconst) {
+                    args = [code.co_consts[intArg]]
+                } else if (opcode in dis.hasfree) {
+                    if (intArg < code.co_cellvars.length) {
+                        args = [code.co_cellvars[intArg]]
+                    } else {
+                        var var_idx = intArg - code.co_cellvars.length
+                        args = [code.co_freevars[var_idx]]
+                    }
+                } else if (opcode in dis.hasname) {
+                    args = [code.co_names[intArg]]
+                } else if (opcode in dis.hasjrel) {
```

Comment:
At column 29: [(no-redeclare) 'var_idx' is already defined.](http://eslint.org/docs/rules/no-redeclare)